v2/src/tupelo.ts

```diff
@@ -0,0 +1,113 @@
+declare const Go: any;
+
+import CID from 'cids';
+
+import * as go from "./js/go"
+import { Transaction } from 'tupelo-messages'
+import {IBlockService, IBlock} from './chaintree/dag/dag'
+import ChainTree from './chaintree/chaintree';
+import { CurrentState } from 'tupelo-messages/signatures/signatures_pb';
+import { generateKeyPair } from 'crypto';
+import { EcdsaKey } from './crypto';
+
+
+export interface IPubSub {
+    publish(topic: string, data: Uint8Array, cb: Function): null
+    subscribe(topic: string, onMsg: Function, cb: Function): null
+}
+
+interface IPlayTransactionOptions {
+    publisher: IPubSub,
+    blockService: IBlockService, 
+    privateKey: Uint8Array,
+    tip: CID, 
+    transactions: Uint8Array[],
+}
+
+class UnderlyingWasm {
+    _populated: boolean;
+
+    constructor() {
+        this._populated = false;
+    }
+
+    generateKey(): Promise<Uint8Array[]> {
+        return new Promise<Uint8Array[]>((res, rej) => { }) // replaced by wasm
+    }
+    newEmptyTree(store: IBlockService, publicKey: Uint8Array): Promise<CID> {
+        return new Promise<CID>((res,rej) => {}) // replaced by wasm
+    }
+    playTransactions(opts: IPlayTransactionOptions): Promise<Uint8Array> {
+        return new Promise<Uint8Array>((res, rej) => { }) // replaced by wasm
+    }
+}
+
+namespace TupeloWasm {
+    const _tupelowasm = new UnderlyingWasm();
+
+    export async function get() {
+        if (_tupelowasm._populated) {
+            return _tupelowasm;
+        }
+
+        go.run("./main.wasm");
+        await go.ready();
+        go.populate(_tupelowasm);
```

Comment:
when go.populate is called is passes this _tupelowasm instance to the "populateLibrary" function which then replaces the methods defined above in UnderlyingWasm